public/app/plugins/datasource/cloud-monitoring/components/GraphPeriod.tsx

```diff
@@ -0,0 +1,45 @@
+import React, { FunctionComponent } from 'react';
+
+import { Input, Switch } from '@grafana/ui';
+
+import { SELECT_WIDTH } from '../constants';
+
+import { QueryEditorRow } from '.';
+
+export interface Props {
+  refId: string;
+  onChange: (period: string) => void;
+  graphPeriod?: string;
+}
+
+export const GraphPeriod: FunctionComponent<Props> = ({ refId, onChange, graphPeriod }) => {
+  return (
+    <>
+      <QueryEditorRow
+        label="Graph period"
+        htmlFor={`${refId}-graph-period`}
+        tooltip={
+          <>
+            Set <code>graph_period</code> which forces a preferred period between points. Automatically set to the
+            current interval if left blank.
+          </>
+        }
+      >
+        <Switch
+          data-testid={`${refId}-switch-graph-period`}
+          value={graphPeriod !== 'disabled'}
+          onChange={(e) => onChange(e.currentTarget.checked ? '' : 'disabled')}
+        />
+        <Input
+          label="value"
```

Comment:
How are they supposed to use this field? Milliseconds? Durations? A tooltip would be helpful.